WooCommerce/Classes/ViewModels/Order Details/Shipping Labels/ShippingLabelPaperSizeOptionView.swift

```diff
@@ -0,0 +1,65 @@
+import Networking
+import SwiftUI
+
+/// Displays a title and image for a shipping label paper size option (e.g. legal, label, letter).
+struct ShippingLabelPaperSizeOptionView: View {
+    struct ViewModel {
+        let title: String
+        let image: Image
+    }
+
+    private let viewModel: ViewModel
+
+    init(paperSize: ShippingLabelPaperSize) {
+        self.viewModel = {
+            let title: String
+            let image: Image
+            switch paperSize {
+            case .label:
+                title = Localization.labelSizeTitle
+                image = PaperSizeImage.label
+            case .legal:
+                title = Localization.legalSizeTitle
+                image = PaperSizeImage.legal
+            case .letter:
+                title = Localization.letterSizeTitle
+                image = PaperSizeImage.letter
+            }
+            return .init(title: title, image: image)
```

Comment:
I have almost 0 experience in `SwiftUI` but it does feel a tad bit weird to create a VM in a View, is that the correct way to go? 😅

Reply:
Do you think it'd be better just to pass the configurable UI properties to the initializer, or a different name like the equivalent concept "Props" in React? Open to suggestions @woocommerce/ios-developers  😄

Reply:
I guess it's weird because in a fully `SwiftUI` world there are not VCs(right?) so this kind of code needs to go somewhere, but since we are mixing `UIKit` and `SwiftUI` I'm not totally sure how the responsibilities should be split 🤔 

In this specific case, I guess it does not hurt much since the `VM` is simple and does not contain any business logic, but I would also want to hear what the more experienced members think!

Reply:
I think that, as Ernesto mentioned, a view should not try to instantiate its own view model. 

ShippingLabelPaperSizeOptionView is instantiated by ShippingLabelPaperSizeOptionsView (by the way, it took me a while to figure out where those two names differ, maybe there is a better way to name one of the two, so that name are more clearly differentiable). 

So, if we wanted to pass a view model to ShippingLabelPaperSizeOptionView, the parent view would have to either a) create that view model, which would put us in the same position, or b) get injected a view model that can spin a view model for a single paper size.

Now, ShippingLabelPaperSizeOptionView creates the view model to encapsulate the logic that decides which title and which image it should render. 

If we encapsulated that logic in an extension to ShippingLabelPaperSize that was declared in the UI layer (something like):

```
protocol ShippingLabelPaperSIzePresentable {
var title: String { get }
var image: Image { get }
}
extension ShippingLabelPaperSize:  ShippingLabelPaperSIzePresentable {
var title: String { 
switch self.label
.....
}
}
```

We would be able to remove that logic from the UI.

Now, the next step would be that we could still initialize ShippingLabelPaperSizeOptionView passing it a ShippingLabelPaperSize or we could choose to declare the reference to  ShippingLabelPaperSize as a state variable in the parent's view. But that might be too much.

So, my suggestion would be:
1. Encapsulate the logic to decide which title and which image to use in an extension to ShippingLabelPaperSize declared in the UI layer, and that acts as kind of a presenter for ShippingLabelPaperSize
2. Pass only an instance of ShippingLabelPaperSize to ShippingLabelPaperSizeOptionView
3. Let ShippingLabelPaperSizeOptionView consume the methods/vars in that extension 

Wha do you think @jaclync ?

Reply:
Thanks both for the feedback! I agree that `ViewModel` is quite odd in SwiftUI, since we generally use it in the "view controller - view model" context in UIKit. If we ever encounter a use case when the number of configurable properties is large, we can come up with some other name specific to the use case.

What do you think of just removing the `ViewModel` and instantiating the `title` and `image` in the initializer like in https://github.com/woocommerce/woocommerce-ios/pull/3322/commits/2456de1fe0a325d8845837a0f85ef79c39f4a188?

Reply:
> ShippingLabelPaperSizeOptionView is instantiated by ShippingLabelPaperSizeOptionsView (by the way, it took me a while to figure out where those two names differ, maybe there is a better way to name one of the two, so that name are more clearly differentiable).

I renamed `ShippingLabelPaperSizeOptionsView` to `ShippingLabelPaperSizeOptionListView` in https://github.com/woocommerce/woocommerce-ios/pull/3322/commits/b2215b500e55f8ee6a3148d7502106ca5ee4d8ee, hopefully they are more distinguishable now 😄